js/http.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+import { Closer } from "./io";
+import * as msg from "gen/msg_generated";
+import { assert, log } from "./util";
+import * as dispatch from "./dispatch";
+import * as flatbuffers from "./flatbuffers";
+import { close } from "./files";
+
+// TODO Cannot use Headers due to bug in ts_declaration_builder.
+// import { Headers } from "./headers";
+// import * as headers from "./headers";
+// import * as domTypes from "./dom_types";
+
+type HttpHandler = (req: ServerRequest, res: ServerResponse) => void;
+
+export class HttpServer implements Closer {
+  private closing = false;
+
+  constructor(readonly rid: number) {
+    assert(rid >= 2);
+  }
+
+  async serve(handler: HttpHandler): Promise<void> {
```

Comment:
It resolves when the server stops serving. This could be useful ... but also it corresponds to a concept on the Rust side that we may or may not want to bind to at some points so I think it makes sense to have it here.

Regarding multiple handlers, ya probably this function shouldn't be called more than once. I consider the surface API a WIP - this PR is just trying to setup the pipes to this.

Reply:
Shouldn't this be > 2 ?